### ISSUE TITLE
some error happen on WSL2 and Java extension

### ISSUE THREAD

**Issue by MajorTomMan** [redhat-developer/vscode-java#2961]
[provide a description of the issue]
When I open my project and wait for Opening Java Projects to finish running, 
![image](https://user-images.githubusercontent.com/47267913/221216325-3016da93-b264-4e93-a7b2-633f50f0a2ab.png)

Java Projects works fine, but the Spring extension has the following error after JavaProjects is loaded
![image](https://user-images.githubusercontent.com/47267913/221216125-da654731-ff15-4074-87c5-091db4084f84.png)
and dont have any output on dashboard
![image](https://user-images.githubusercontent.com/47267913/221217527-ab563d0f-24e8-4818-959d-a589a140ae6f.png)
**This error only occurs when running Java projects in Vscode on WSL2 and I have repeatedly reinstalled Java dependencies and Spring dependencies several times..........**

**here is some logs i got**
!ENTRY org.eclipse.jdt.ls.core 4 2 2023-02-24 23:17:09.968
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.jdt.ls.core".
!STACK 0
java.lang.NullPointerException: Cannot invoke "java.io.PrintWriter.println(String)" because "this.printwriter" is null
	at org.springframework.tooling.jdt.ls.commons.Logger$DefaultLogger.log(Logger.java:51)
	at org.springframework.tooling.jdt.ls.extension.ClasspathListenerHandler.executeCommand(ClasspathListenerHandler.java:48)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:559)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

!ENTRY org.eclipse.jdt.ls.core 4 0 2023-02-24 23:17:09.969
!MESSAGE Error in calling delegate command handler
!STACK 0
java.lang.NullPointerException: Cannot invoke "java.io.PrintWriter.println(String)" because "this.printwriter" is null
	at org.springframework.tooling.jdt.ls.commons.Logger$DefaultLogger.log(Logger.java:51)
	at org.springframework.tooling.jdt.ls.extension.ClasspathListenerHandler.executeCommand(ClasspathListenerHandler.java:48)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:559)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
##### Environment
- Operating System: Ubuntu 20.04.5 LTS
- JDK version: 
        openjdk version "17.0.5" 2022-10-18
        OpenJDK Runtime Environment (build 17.0.5+8-Ubuntu-2ubuntu120.04)
        OpenJDK 64-Bit Server VM (build 17.0.5+8-Ubuntu-2ubuntu120.04, mixed mode, sharing)
- Visual Studio Code version: 1.75.1
- Java extension version: v1.15.0

##### Steps To Reproduce
1. [step 1]
   Waiting for the loading is done 
2. [step 2]
   click spring extension after step 1 is done 
2. [step 3]
   open the console and see some error report

**Comment by fbricon** [comment]
This issue is out of scope of vscode-java. Please open a ticket on the Spring Tool Suite repo: https://github.com/spring-projects/sts4/issues

cc @martinlippert
